app/controllers/mentor/applications_controller.rb

```diff
@@ -8,6 +8,20 @@ def show
     @comment     = application.find_or_initialize_comment_by(current_user)
   end
 
+  def signoff
+    @application = Application.find(application.id)
+    @application.sign_off! as: current_user
+    flash[:notice] = "Successfully signed-off #{@application.team.name}'s application."
```

Comment:
Not sure about how smart Rails is about having already queried for the `team`. To play save, I'd suggest to rather use
```ruby
"#{application.team_name}"
```
here, since this won't need an additional query.

Same for the flash in `fav` of course.